### PR TITLE
Fixed issue related to xfs_growfs to resize.

### PR DESCRIPTION
**Pull request by sbulage**
Signed-off-by: Satyajit Bulage <sbulage@redhat.com>

##### SUMMARY

This PR will fix resizing partition by using `xfs_growfs` command.
Fixed Issue #63278

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

filesystem
##### ADDITIONAL INFORMATION

```paste below
ansible -i inventory -m filesystem -a 'dev=/dev/fedora_localhost-live/mylv fstype=xfs resizefs=true mount_dir=/test' all -v
Using /opt/git_repos/ansible/ansible/ansible.cfg as config file
[DEPRECATION WARNING]: Distribution fedora 28 on host 192.168.122.139 should use /usr/bin/python3, but is using /usr/bin/python for 
backward compatibility with prior Ansible releases. A future Ansible release will default to using the discovered platform python for 
this host. See https://docs.ansible.com/ansible/devel/reference_appendices/interpreter_discovery.html for more information. This 
feature will be removed in version 2.12. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.

192.168.122.139 | CHANGED => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    }, 
    "changed": true, 
    "invocation": {
        "module_args": {
            "dev": "/dev/fedora_localhost-live/mylv", 
            "force": false, 
            "fstype": "xfs", 
            "mount_dir": "/test", 
            "opts": null, 
            "resizefs": true
        }
    }, 
    "msg": "meta-data=/dev/mapper/fedora_localhost--live-mylv isize=512    agcount=4, agsize=98304 blks\n         =                       sectsz=512   attr=2, projid32bit=1\n         =                       crc=1        finobt=1 spinodes=0 rmapbt=0\n         =                       reflink=0\ndata     =                       bsize=4096   blocks=393216, imaxpct=25\n         =                       sunit=0      swidth=0 blks\nnaming   =version 2              bsize=4096   ascii-ci=0 ftype=1\nlog      =internal               bsize=4096   blocks=2560, version=2\n         =                       sectsz=512   sunit=0 blks, lazy-count=1\nrealtime =none                   extsz=4096   blocks=0, rtextents=0\ndata blocks changed from 393216 to 521216\n"
}

```

### CODE REVIEW

**Comment by sbulage** [comment]
@Akasurde Please look into this.

File: lib/ansible/modules/system/filesystem.py [hunk 1]
```diff
@@ -56,6 +56,9 @@
   opts:
     description:
     - List of options to be passed to mkfs command.
+  mount_dir:
+    description:
+    - To grow XFS filesystem mount point required.
```
Comment (Akasurde) [inline lib/ansible/modules/system/filesystem.py:61]:
```suggestion
    - To grow XFS filesystem mount point required.
    version_added: "2.10"
```

File: lib/ansible/modules/system/filesystem.py [hunk 2]
```diff
@@ -350,6 +355,7 @@ def main():
             opts=dict(),
             force=dict(type='bool', default=False),
             resizefs=dict(type='bool', default=False),
+            mount_dir=dict(default=False)
```
Comment (Akasurde) [inline lib/ansible/modules/system/filesystem.py]:
```suggestion
            mount_dir=dict()
```

default=False is default so no need to add.

File: lib/ansible/modules/system/filesystem.py [hunk 3]
```diff
@@ -139,19 +142,21 @@ def grow_cmd(self, dev):
...
         except NotImplementedError:
             self.module.fail_json(changed=False, msg="module does not support resizing %s filesystem yet." % self.fstype)
 
         if not fssize_in_bytes < devsize_in_bytes:
             self.module.exit_json(changed=False, msg="%s filesystem is using the whole device %s" % (self.fstype, dev))
         elif self.module.check_mode:
             self.module.exit_json(changed=True, msg="Resizing filesystem %s on device %s" % (self.fstype, dev))
+        elif self.fstype == 'XFS':
+            _, out, _ = self.module.run_command(self.grow_cmd(dev=mount_dir), check_rc=True)
         else:
```
Comment (Akasurde) [inline lib/ansible/modules/system/filesystem.py:159]: I would write:

```python
else:
   if self.fstype == 'XFS':
   ...
   else:
   ...
   return out
```

**Review by Akasurde — CHANGES_REQUESTED** [review]
@sbulage Thanks for the contribution. Could you please add a unit test for this change [here](https://github.com/ansible/ansible/tree/devel/test/units/module_utils/basic/test_filesystem.py) ?

**Comment by sbulage** [comment]
@Akasurde Thanks for the quick suggestions. I will add unit test for it.

File: lib/ansible/modules/system/filesystem.py (hunk 1 above)
Comment (sbulage) [reply lib/ansible/modules/system/filesystem.py:61]: Added.

File: lib/ansible/modules/system/filesystem.py (hunk 2 above)
Comment (sbulage) [reply lib/ansible/modules/system/filesystem.py]: Default value removed.

**Comment by holdenSK** [comment]
Same issue is being solved in  ansible-collections/community.general#137

**Comment by anatolinicolae** [comment]
🤔

**Comment by Akasurde** [comment]
Superseded by https://github.com/ansible-collections/community.general/pull/137

**Comment by Akasurde** [comment]
@sbulage Thanks for the contribution.

**Pull request closed without merging**